[Acerizm] iP

Acerizm opened the pull request:
##### DukePro[](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html#dukepro)

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

-   text-based
-   easy to learn
-   FAST  _SUPER_  FAST to use

All you need to do is,

1.  download it from  [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2.  double-click it.
3.  add your tasks.
4.  let it manage your tasks for you 😉

And it is  **FREE**!

Features:

-   Managing tasks
-   Managing deadlines (coming soon)
-   Reminders (coming soon)

----------

If you are a Java programmer, you can use it to practice Java too. Here's the  `main`  method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Acerizm committed:
Added custom logo and basic chat feature as per requirements of of Level-1

Acerizm committed:
added list for printing out stuff

Acerizm committed:
fixed bug for listing incorrect index

Acerizm committed:
added Task class and added enums as well

Acerizm committed:
added typeOfTasks as enums

Acerizm committed:
refactoring code to support inheritance for tasks

Acerizm committed:
fixed major bugs and output is showing correctly as manually tested without automated testing

Acerizm committed:
tested testing using bat file

Acerizm committed:
added DukeException class with proper handling of error using custom exception class and methods. Refactored code to fit level 5.

Acerizm committed:
added delete capability with custom error validation

Acerizm committed:
add saving and loading data using .txt file

Acerizm committed:
add dates and times for events and deadlines

Acerizm committed:
Merge branch 'branch-Level-7'

Acerizm committed:
fix branching issue

Acerizm committed:
seperate logic into different classes

c0j0s commented on src/main/java/Acerizm.java:
```diff
@@ -0,0 +1,289 @@
+                    // for deadline
+                    System.out.println("*-".repeat(100));
+                    String userInput = convertToUserInput(input,TypeOfTask.deadline,"");
+                    // added additional variable to store the date of the deadline
+                    String[] dateTime = convertToUserInput(input,TypeOfTask.deadline,"/by").split(" ");
+                    Task newTask = new Deadline(userInput,dateTime[0],dateTime[1]);
+                    taskList.add(newTask);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(newTask.toString());
+                    System.out.println(String.format("Now you have %d tasks in the list",taskList.size()));
+                    System.out.println("*-".repeat(100));
+                } else if (actionTaken == TypeOfTask.event) {
+                    // for event
+                    System.out.println("*-".repeat(100));
+                    String userInput = convertToUserInput(input,TypeOfTask.event,"");
+                    // added additional variable to store the start and end time of event
+                    String startTime = convertToUserInput(input,TypeOfTask.event,"/from");
+                    String endTime = convertToUserInput(input,TypeOfTask.event,"/to");
+                    Task newTask = new Event(userInput,startTime,endTime);
+                    taskList.add(newTask);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(newTask.toString());
+                    System.out.println(String.format("Now you have %d tasks in the list",taskList.size()));
+                    System.out.println("*-".repeat(100));
+
+                } else if(actionTaken == TypeOfTask.delete) {
+                    int taskIndexDelete = Integer.parseInt(convertToUserInput(input,TypeOfTask.delete,"")) - 1;
+                    Task taskToBeDeleted = taskList.get(taskIndexDelete);
+                    System.out.println("Noted! I've removed this task:");
+                    System.out.println(String.format("%d. %s",taskIndexDelete+1,taskToBeDeleted.toString()));
+                    taskList.remove(taskIndexDelete);
+                    System.out.println(String.format("Now you have %d tasks in the list",taskList.size()));
+                    System.out.println("*-".repeat(100));
+                }else {
+                    // if it doesn't match any action, do nothing
+                    // print error
+                    // this else clause will never be reached as we have handled the exceptions in DukeExceptions
+                    System.out.println("Oh no! You forgot to type in something useful :p");
+                    System.out.println("What can I do for you?");
+                }
```
Nice, neat and clean. You can use a switch case too.

c0j0s commented on src/main/java/Acerizm.java:
```diff
@@ -0,0 +1,289 @@
+// import libraries here
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+// for serialization and deserialization
+import java.io.FileOutputStream;
+import java.io.ObjectOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
```
Very organise👍🏻

c0j0s commented on src/main/java/Acerizm.java:
```diff
@@ -0,0 +1,289 @@
+        of task.
+     */
+    // limiter -> "/by", "/from", "/to"
+    public static String convertToUserInput(String[] input, TypeOfTask action,String limiter ) throws DukeException{
```
perhaps you might want to standardise your white spaces.

c0j0s commented on src/main/java/Acerizm.java:
```diff
@@ -0,0 +1,289 @@
+     */
+    // limiter -> "/by", "/from", "/to"
+    public static String convertToUserInput(String[] input, TypeOfTask action,String limiter ) throws DukeException{
+        switch(action){
+            case todo: {
```
Switch and case statement should be on the same indentation level?

c0j0s commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+    public String toString() {
+        //return "[D]" + super.getDescription() + " (by: " + this.day + ")";
+        String day = this.getDate().format(DateTimeFormatter.ofPattern(DESIRED_DATE_FORMAT));
+        return String.format("[D][%s] %s (by: %s %s)",super.getStatusIcon(),super.getDescription(),day,this.getTime().toString());
```
Perhaps a shorter statement?

c0j0s reviewed (commented):
Looking good, I like your coding style, tidy and easy to read 👍🏻

Acerizm committed:
reafctor code with multiple classes

Acerizm committed:
add packages and fix bugs related to marking and unmarking tasks

Acerizm committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Acerizm committed:
add JUnit test folders and files for testing

Acerizm committed:
add JavaDocs for classes in "actions" package

Acerizm committed:
add JavaDocs for remaining classes in other packages.

Acerizm committed:
update JavaDocs

Acerizm committed:
Revert "update JavaDocs"

This reverts commit b4a1b3e2bd2a86d3f0c89d8122fed30ff23f2765.

Acerizm committed:
Revert "Revert "update JavaDocs""

This reverts commit 4d015f63ed7ddba5af3eaf2682f78b81fab385de.

Acerizm committed:
add appropiate spacing between variables and function declaration to follow the Java coding standard

Acerizm committed:
add find command that allows user to search for similar task based on description given

weekiat-douze commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,71 @@
+            try {
+                String fullCommand = ui.readCommand();
+                Command command = this.parser.parse(fullCommand);
+                command.execute(this.taskList,this.ui,this.storage);
```
You might have missed out on this line when formatting. "Commas should be followed by a white space."

weekiat-douze replied:
You might have missed out on this line when formatting. "Commas should be followed by a white space."

weekiat-douze commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,35 @@
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        // need to save the updated taskList to the storage
+        // the ui will be responsible in displaying whatever needs to be displayed -> not the responsibility of the command classes
+        ui.displayResult(TypeOfTask.bye,null,null);
```
Spacing here as well.

weekiat-douze commented on src/main/java/duke/action/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+    public String toString() {
+        //return "[D]" + super.getDescription() + " (by: " + this.day + ")";
+        String day = this.getDate().format(DateTimeFormatter.ofPattern(DESIRED_DATE_FORMAT));
+        return String.format("[D][%s] %s (by: %s %s)", super.getStatusIcon(), super.getDescription(), day, this.getTime().toString());
```
Would line wrapping be better for this line?

weekiat-douze commented on src/main/java/duke/action/Event.java:
```diff
@@ -0,0 +1,78 @@
+        } else {
+            String startTime = this.startTime.toString();
+            String endTime = this.endTime.toString();
+            return String.format("[E][%s] %s (from: %s %s to: %s %s)", super.getStatusIcon(), super.getDescription(), this.startDay, startTime, this.endDay, endTime);
```
I think line wrapping would be good here as well.

weekiat-douze commented on src/main/java/duke/action/Task.java:
```diff
@@ -0,0 +1,76 @@
+     * @return character format of task
+     */
+    public String getTypeOfTask() {
+        switch(this.type){
```
Might have missed this switch-case formatting. No indentations for case clauses

weekiat-douze reviewed (commented):
Looks good! All the best!

Acerizm committed:
Merge branch 'branch-Level-9'

Acerizm committed:
add checkstyle configurations

Acerizm committed:
refactor most code to fit with checkstyle

Acerizm committed:
add Launcher for JavaFX

Acerizm committed:
add AnchorPane for JavaFX

Acerizm committed:
add dialog boxes

Acerizm committed:
add basic GUI

Acerizm committed:
modify GUI with FXML

Acerizm committed:
refactor all commands to return result for GUI

Acerizm committed:
add custom styling and quit button with CSS

Acerizm committed:
add assertion for filepath used for storage

Acerizm committed:
refactor errors in DukeException into seperate error classes

Acerizm committed:
update entire codebase to closely ensure that code is of quality

Acerizm committed:
update add task function to check for duplicates

Acerizm committed:
Merge pull request #4 from Acerizm/branch-A-CodeQuality

updates all code to comply with coding standards

Acerizm committed:
Merge branch 'master' into branch-C-DetectDuplicates

Acerizm committed:
update storage error to show duplicate task error when user enters the same task

Acerizm committed:
Merge pull request #5 from Acerizm/branch-C-DetectDuplicates

update add task function to check for duplicates

Acerizm committed:
update UI of program

There are multiple edits of background color to differentiate between user and duke.
The images used are modified to have round edges using Circles

Adding circle nodes to make images have round borders.

Acerizm committed:
remove unnecessary imports in MainWindow.java

Acerizm committed:
modify color theme of program

Acerizm committed:
add more personalisation

Acerizm committed:
modify generic error message

Acerizm committed:
update dialog box to have infinite height and fix bugs for todo command

Acerizm committed:
add image of main page

Acerizm committed:
add user guide

Acerizm committed:
update main readme

Acerizm committed:
remove generated javadoc site

Acerizm committed:
update emojis used in readme files

Acerizm committed:
add whitespace after last example for event description

Acerizm committed:
add support for .jar file
